pkg/utils/report/results.go

```diff
@@ -86,72 +87,76 @@ func SeverityFromString(severity string) policyreportv1alpha2.PolicySeverity {
 	return ""
 }
 
-func EngineResponseToReportResults(response engineapi.EngineResponse) []policyreportv1alpha2.PolicyReportResult {
-	pol := response.Policy()
-	var results []policyreportv1alpha2.PolicyReportResult
-	if pol.GetType() == engineapi.KyvernoPolicyType {
-		key, _ := cache.MetaNamespaceKeyFunc(pol.AsKyvernoPolicy())
-		for _, ruleResult := range response.PolicyResponse.Rules {
-			annotations := pol.GetAnnotations()
-			result := policyreportv1alpha2.PolicyReportResult{
-				Source:  kyverno.ValueKyvernoApp,
-				Policy:  key,
-				Rule:    ruleResult.Name(),
-				Message: ruleResult.Message(),
-				Result:  toPolicyResult(ruleResult.Status()),
-				Scored:  annotations[kyverno.AnnotationPolicyScored] != "false",
-				Timestamp: metav1.Timestamp{
-					Seconds: time.Now().Unix(),
-				},
-				Category: annotations[kyverno.AnnotationPolicyCategory],
-				Severity: SeverityFromString(annotations[kyverno.AnnotationPolicySeverity]),
-			}
-			if ruleResult.Exception() != nil {
-				result.Properties = map[string]string{
-					"exception": ruleResult.Exception().Name,
-				}
-			}
-			pss := ruleResult.PodSecurityChecks()
-			if pss != nil {
-				var controls []string
-				for _, check := range pss.Checks {
-					if !check.CheckResult.Allowed {
-						controls = append(controls, check.ID)
-					}
-				}
-				if len(controls) > 0 {
-					sort.Strings(controls)
-					result.Properties = map[string]string{
-						"standard": string(pss.Level),
-						"version":  pss.Version,
-						"controls": strings.Join(controls, ","),
-					}
-				}
-			}
-			if result.Result == "fail" && !result.Scored {
-				result.Result = "warn"
-			}
-			results = append(results, result)
+func addProperty(k, v string, result *policyreportv1alpha2.PolicyReportResult) {
+	if result.Properties == nil {
+		result.Properties = map[string]string{}
+	}
+
+	result.Properties[k] = v
+}
+
+func ToPolicyReportResult(policyType engineapi.PolicyType, policyName string, ruleResult engineapi.RuleResponse, annotations map[string]string, resource *corev1.ObjectReference) policyreportv1alpha2.PolicyReportResult {
+	result := policyreportv1alpha2.PolicyReportResult{
+		Source:  kyverno.ValueKyvernoApp,
+		Policy:  policyName,
+		Rule:    ruleResult.Name(),
+		Message: ruleResult.Message(),
+		Result:  toPolicyResult(ruleResult.Status()),
+		Scored:  annotations[kyverno.AnnotationPolicyScored] != "false",
```

Comment:
This annotation is specific to Kyverno policies. It should be unset for VAPs.

Reply:
If the annotation is not present, it will default to true, right? Would that be a problem.